### ISSUE TITLE
Migrate the Setting page into private group, e.g. `topgear`

### ISSUE THREAD

**Issue by LieutenantRoger** [topcoder-platform/community-app#4088]
**Describe the bug**
Migrate the Setting page into private group: `topgear`

**To Reproduce**
1. Go to `https://topgear.topcoder-dev.com/members/TonyJ` 
2. Hover the user `TonyJ` in the top right area of the header to show a sub-header
3. Click on 'SETTINGS' under the sub-header
4. The site will navigate user to `TonyJ` public setting screen right now.

**Expected behavior**
We need to migrate current SETTINGS page into private domain.

So after step 3, user should be remain in the private group with Setting section displayed, and the header should be still the Private domain header.

**Additional context**

The work should be down against, `feature-private-settings`

**Comment by LieutenantRoger** [comment]
Hi @Oanh-and-only-Oanh  

Here's a challenge spec for this ticket, please take a look:  https://docs.google.com/document/d/1-Zwb5QGPm2ByzAb_HE9LKtiEEefPScTCJA5HGSxvID4/edit?usp=sharing

**Comment by Oanh-and-only-Oanh** [comment]
putting this on hold for now.

**Comment by LieutenantRoger** [comment]
In this ticket, we plan to support these private groups firstly to have `Settings` page:

https://topgear.topcoder-dev.com/challenges
https://cs.topcoder-dev.com/challenges
https://veterans.topcoder-dev.com/challenges
https://blockchain.topcoder-dev.com/challenges?communityId=blockchain
https://cognitive.topcoder-dev.com/challenges?communityId=cognitive

**Comment by LieutenantRoger** [comment]
Hi @Oanh-and-only-Oanh 

Could you tell why the deployed code in community#dev environment and my local deployed version are different for cs community ?   https://take.ms/BJid1

Actually I tried otherwise private groups as well. 

Right now, it seems only topgear private group are the same between deployed locally and community#dev environment.  https://take.ms/xq4Ou
@SathyaJayabal  has confirmed `topgear` private group setting page is working fine.

I suspect my changes to otherwise private groups  not getting executed due to the route load balance in the deployed environment.

**Comment by Oanh-and-only-Oanh** [comment]
Hi @LieutenantRoger, I'm not sure about the why there's a difference between CS dev and your local version. I can look into it. Until then, let's narrow down the scope so excludes any private community with this issue. As long as we can get this complete for Topgear...

**Comment by LieutenantRoger** [comment]
Sure

**Comment by SathyaJayabal** [comment]
Verified for Topgear
<img width="1440" alt="Screenshot 2020-03-26 at 8 53 37 AM" src="https://user-images.githubusercontent.com/58783823/77607296-decfbb00-6f3f-11ea-861d-04737bd02623.png">

**Comment by LieutenantRoger** [comment]
Thanks

**Comment by lakshmiathreya** [comment]
As seen on Beta with wipro login:
<img width="1664" alt="Screenshot 2020-03-26 at 12 27 49 PM" src="https://user-images.githubusercontent.com/51733042/77619059-6c6dd380-6f5d-11ea-8e94-701d94b4fbdd.png">

**Comment by lakshmiathreya** [comment]
Settings link from the Profile redirects to Prod. Only the direct url shown above loads the Beta Settings page.

**Comment by lakshmiathreya** [comment]
All Settings sanity checked for this Profile on Beta. The changes seem to be set correctly.

**Comment by LieutenantRoger** [comment]
> Settings link from the Profile redirects to Prod. Only the direct url shown above loads the Beta Settings page.

Hi @lakshmiathreya  yes, it was due to the `beta environment`, when we use topgear.topcoder.com to access the topgear group, the navigation will work correctly.

**Comment by lakshmiathreya** [comment]
Verified on topgear.topcoder.com via Wipro login.
